libs/ui-lib/lib/ocm/components/clusterConfiguration/ControlPlaneNodesDropdown.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import { FormGroup, Tooltip } from '@patternfly/react-core';
+import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core/deprecated';
+import { useField } from 'formik';
+import { getFieldId, PopoverIcon } from '../../../common';
+import {
+  NewFeatureSupportLevelMap,
+  useNewFeatureSupportLevel,
+} from '../../../common/components/newFeatureSupportLevels';
+import OcmSNODisclaimer from './OcmSNODisclaimer';
+
+const INPUT_NAME = 'controlPlaneCount';
+const fieldId = getFieldId(INPUT_NAME, 'input');
+const DEFAULT_VALUE = '3';
+
+export const ControlPlaneNodesLabel = () => {
+  return (
+    <>
+      Number of control plane nodes <PopoverIcon bodyContent={<>Info control plane nodes</>} />
+    </>
+  );
+};
+
+interface ControlPlaneNodesOption {
+  value: string;
+  label: string;
+}
+
+interface ControlPlaneNodesDropdownProps {
+  openshiftVersion: string;
+  cpuArch: string;
+  platform: string;
+  featureSupportLevelData?: NewFeatureSupportLevelMap;
+}
+
+const isDropdownItemEnabled = (
+  controlPlaneNodeCount: string,
+  openshiftVersion: string,
+  cpuArch: string,
+  platform: string,
+): boolean => {
+  if (controlPlaneNodeCount === '4' || controlPlaneNodeCount === '5') {
+    return (
+      parseFloat(openshiftVersion) >= 4.18 &&
+      cpuArch === 'x86_64' &&
+      (platform === 'none' || platform === 'baremetal')
```

Comment:
We are not sure yet whether it will be available for `none` platform, I am still testing it. currently it allows only `baremetal`

Reply:
Maybe we can fetch this from assisted-service feature support endpoint ?

Reply:
We have this info -

```
[root@edge-20 assisted-test-infra]#  curl "http://10.1.178.29:8060/api/assisted-install/v2/support-levels/features?openshift_version=4.18.0-ec.3&cpu_architecture=x86_64&&platform=baremetal" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   932  100   932    0     0   910k      0 --:--:-- --:--:-- --:--:--  910k
{
  "features": {
    "BAREMETAL_PLATFORM": "supported",
    "CLUSTER_MANAGED_NETWORKING": "supported",
    "CNV": "supported",
    "CUSTOM_MANIFEST": "supported",
    "DUAL_STACK": "supported",
    "DUAL_STACK_VIPS": "supported",
    "EXTERNAL_PLATFORM": "supported",
    "EXTERNAL_PLATFORM_OCI": "supported",
    "FULL_ISO": "supported",
    "LSO": "supported",
    "LVM": "supported",
    "MCE": "supported",
    "MINIMAL_ISO": "supported",
    "MTV": "supported",
    "NODE_FEATURE_DISCOVERY": "dev-preview",
    "NONE_PLATFORM": "supported",
    "NON_STANDARD_HA_CONTROL_PLANE": "supported",
    "NUTANIX_INTEGRATION": "supported",
    "NVIDIA_GPU": "dev-preview",
    "ODF": "supported",
    "OPENSHIFT_AI": "dev-preview",
    "OVN_NETWORK_TYPE": "supported",
    "PIPELINES": "dev-preview",
    "SDN_NETWORK_TYPE": "unavailable",
    "SERVERLESS": "dev-preview",
    "SERVICEMESH": "dev-preview",
    "SINGLE_NODE_EXPANSION": "supported",
    "SKIP_MCO_REBOOT": "supported",
    "SNO": "supported",
    "USER_MANAGED_NETWORKING": "supported",
    "VIP_AUTO_ALLOC": "unavailable",
    "VSPHERE_INTEGRATION": "supported"
  }
}
[root@edge-20 assisted-test-infra]#  curl "http://10.1.178.29:8060/api/assisted-install/v2/support-levels/features?openshift_version=4.18.0-ec.3&cpu_architecture=x86_64&&platform=none" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   932  100   932    0     0   910k      0 --:--:-- --:--:-- --:--:--  910k
{
  "features": {
    "BAREMETAL_PLATFORM": "supported",
    "CLUSTER_MANAGED_NETWORKING": "supported",
    "CNV": "supported",
    "CUSTOM_MANIFEST": "supported",
    "DUAL_STACK": "supported",
    "DUAL_STACK_VIPS": "supported",
    "EXTERNAL_PLATFORM": "supported",
    "EXTERNAL_PLATFORM_OCI": "supported",
    "FULL_ISO": "supported",
    "LSO": "supported",
    "LVM": "supported",
    "MCE": "supported",
    "MINIMAL_ISO": "supported",
    "MTV": "supported",
    "NODE_FEATURE_DISCOVERY": "dev-preview",
    "NONE_PLATFORM": "supported",
    "NON_STANDARD_HA_CONTROL_PLANE": "supported",
    "NUTANIX_INTEGRATION": "supported",
    "NVIDIA_GPU": "dev-preview",
    "ODF": "supported",
    "OPENSHIFT_AI": "dev-preview",
    "OVN_NETWORK_TYPE": "supported",
    "PIPELINES": "dev-preview",
    "SDN_NETWORK_TYPE": "unavailable",
    "SERVERLESS": "dev-preview",
    "SERVICEMESH": "dev-preview",
    "SINGLE_NODE_EXPANSION": "supported",
    "SKIP_MCO_REBOOT": "supported",
    "SNO": "supported",
    "USER_MANAGED_NETWORKING": "supported",
    "VIP_AUTO_ALLOC": "unavailable",
    "VSPHERE_INTEGRATION": "supported"
  }
}
```

how does the UI knows if the platform is `baremetal` or `none` before the cluster was registered ? @ammont82

Reply:
btw - none platform will be enabled after all - https://github.com/openshift/assisted-service/pull/7059

Reply:
@danmanor then I can take a look to the feature "NON_STANDARD_HA_CONTROL_PLANE" to allow users to choose 4/5 CPN?

Reply:
Yes